### ISSUE TITLE
java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0

### ISSUE THREAD

**Issue by linchaolong** [PhilJay/MPAndroidChart#3011]
I'm run MPChartExample's Line Chart(Dual YAxis) happen IndexOutOfBoundsException, Log as follows:

```
04-14 11:32:38.199 28637-28637/com.xxmassdeveloper.mpchartexample D/ActivityThreadInjector: clearCachedDrawables.
04-14 11:32:38.224 28637-28655/com.xxmassdeveloper.mpchartexample D/ufoEGL: Pixel Format : GGL_PIXEL_FORMAT_RGBA_8888
04-14 11:32:38.358 28637-28655/com.xxmassdeveloper.mpchartexample D/ufoEGL: Surface destroy: ANDROID_NATIVE_WINDOW_MAGIC
04-14 11:32:42.502 28637-28637/? D/AndroidRuntime: Shutting down VM
04-14 11:32:42.504 28637-28637/? E/AndroidRuntime: FATAL EXCEPTION: main
                                                   Process: com.xxmassdeveloper.mpchartexample, PID: 28637
                                                   java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0
                                                       at java.util.ArrayList.throwIndexOutOfBoundsException(ArrayList.java:255)
                                                       at java.util.ArrayList.get(ArrayList.java:308)
                                                       at com.github.mikephil.charting.data.DataSet.getEntryForIndex(DataSet.java:286)
                                                       at com.github.mikephil.charting.utils.Transformer.generateTransformedValuesLine(Transformer.java:184)
                                                       at com.github.mikephil.charting.renderer.LineChartRenderer.drawValues(LineChartRenderer.java:547)
                                                       at com.github.mikephil.charting.charts.BarLineChartBase.onDraw(BarLineChartBase.java:263)
                                                       at android.view.View.draw(View.java:15237)
                                                       at android.view.View.updateDisplayListIfDirty(View.java:14173)
                                                       at android.view.View.getDisplayList(View.java:14195)
                                                       at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3389)
                                                       at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3368)
                                                       at android.view.View.updateDisplayListIfDirty(View.java:14133)
                                                       at android.view.View.getDisplayList(View.java:14195)
                                                       at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3389)
                                                       at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3368)
                                                       at android.view.View.updateDisplayListIfDirty(View.java:14133)
                                                       at android.view.View.getDisplayList(View.java:14195)
                                                       at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3389)
                                                       at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3368)
                                                       at android.view.View.updateDisplayListIfDirty(View.java:14133)
                                                       at android.view.View.getDisplayList(View.java:14195)
                                                       at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3389)
                                                       at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3368)
                                                       at android.view.View.updateDisplayListIfDirty(View.java:14133)
                                                       at android.view.View.getDisplayList(View.java:14195)
                                                       at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:273)
                                                       at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:279)
                                                       at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:318)
                                                       at android.view.ViewRootImpl.draw(ViewRootImpl.java:2552)
                                                       at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:2374)
                                                       at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2004)
                                                       at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1083)
                                                       at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:5917)
                                                       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:769)
                                                       at android.view.Choreographer.doCallbacks(Choreographer.java:582)
                                                       at android.view.Choreographer.doFrame(Choreographer.java:552)
                                                       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:755)
                                                       at android.os.Handler.handleCallback(Handler.java:739)
                                                       at android.os.Handler.dispatchMessage(Handler.java:95)
                                                       at android.os.Looper.loop(Looper.java:135)
                                                       at android.app.ActivityThread.main(ActivityThread.java:5275)
                                                       at java.lang.reflect.Method.invoke(Native Method)
                                                       at java.lang.reflect.Method.invoke(Method.java:372)
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:912)
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:707)
04-14 11:32:42.509 28637-28637/? I/Process: Sending signal. PID: 28637 SIG: 9
```

**Comment by chuzhmarnyv** [comment]
It is bug in last release, you can check v3.0.1 it should work fine

**Comment by chuzhmarnyv** [comment]
#2958 fix

**Comment by TheLordBaski** [comment]
@chuzhmarnyv  Tried version 3.0.1, still not working. Had to rollback to version 2.*.

**Comment by liu4113231988** [comment]
I have the same problem

**Comment by MGaetan89** [comment]
This is a duplicate of #2450

**Comment by MohammadAliK** [comment]
i have also this problem

**Comment by almic** [comment]
fixed now in d67ea481af1e8528c617dfd404d0c7827b0134b2

**Comment by AbdubositKhasanov** [comment]
zb
